lib/workflow/postgres/postgres.go

```diff
@@ -412,6 +412,56 @@ VALUES ($1, $2, $3, $4, $5, $6)`,
 	return wrapErr(tx.Commit(), "failed to commit workflow schedule update")
 }
 
+func (pg *PostgresBackend) GetWorkflowInstanceByName(ctx context.Context, instanceName string, workflowName string) (*backend.WorkflowInstance, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	tx, err := pg.db.BeginTx(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	row := tx.QueryRowContext(ctx,
+		"SELECT status, parameters, payload FROM workflow_instances WHERE workflow_name = $1 AND instance_name = $2",
+		workflowName, instanceName,
+	)
+	workflowInstance := backend.WorkflowInstance{
+		WorkflowName: workflowName,
+		InstanceName: instanceName,
+	}
+	err = row.Scan(
+		&workflowInstance.Status,
+		&workflowInstance.Parameters,
+		&workflowInstance.Payload,
+	)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			row := tx.QueryRowContext(ctx,
+				"SELECT parameters, result FROM workflow_results WHERE workflow_name = $1 AND instance_name = $2 ORDER BY end_at DESC",
```

Comment:
I'm a little hesitant to use the `end_at` time here.  I wonder, since for a given named workflow, we know that we can't have overlapping executions of it, might we instead order by the ID serial?  We'll have a potential rollover issue then. 🤷‍♂ Perhaps depend on the time isn't too bad.

Reply:
i like that better. let me give it a go